### PR TITLE
[functionapp] - add ability to create functionapp from acr

### PR DESCRIPTION
**Pull request by ankitkumarr**
Need to confirm the environment variable settings with @balag0 

\cc @panchagnula @dariagrigoriu @ahmedelnably 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### CODE REVIEW

**Comment by marstr** [comment]
CI may be blocked by #8414 at the moment. Once it is merged, try rebasing on-top of upstream/dev and trying again.

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py [hunk 1]
```diff
@@ -215,6 +215,10 @@ def load_arguments(self, _):
             c.argument('generic_configurations', nargs='+',
                        help='provide site configuration list in a format of either "key=value" pair or "@<json_file>"')
 
+        with self.argument_context(scope + ' deployment container config') as c:
+            c.argument('enable', options_list=['--enable-cd', '-e'], help='enable/disable continuous deployment',
+                       arg_type=get_enum_type(['true', 'false']))
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py]: suggest `get_three_state_flag()`

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py [hunk 2]
```diff
@@ -384,3 +386,9 @@ def load_arguments(self, _):
         c.argument('sku', required=True, help='The SKU of the app service plan.')
         c.argument('number_of_workers', help='The number of workers for the app service plan.')
         c.argument('tags', arg_type=tags_type)
+
+    with self.argument_context('functionapp config container') as c:
+        c.argument('docker_registry_server_url', options_list=['--docker-registry-server-url', '-r'], help='the container registry server url')
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py]: please reuse the same params of `webapp config container`

**Comment by panchagnula** [comment]
@Nking92 can you look at this PR as well, given you are working on web apps & ACR support :)

**Comment by Nking92** [comment]
Looks good

**Pull request merged**